Add a Description field for more detailed documentation of tasks

stansey opened the pull request:
example output: 

```
$ taskrunner -describe
...
    mytask
    mytask/build	 Generates client bundle. Compiles typescript.
```

I tested this by adding a `Description` field to a task called `mytask/build`

I added a -describe option instead of modifying -list because I wanted to maintain fzf (fuzzy find) compatibility. I tried running taskrunner commands with fzf using -describe instead of -list. No matter what punctuation I wrapped the Description with, taskrunner always tried to run every word in the description as a task. I tried # and quotes and semi-colons.

berfarah commented on task.go:
```diff
@@ -8,6 +8,8 @@ import (
 
 type Task struct {
 	Name string
+	Description string // A one or two sentence description of what the task does
```
nit: Can we add this comment above the field?

berfarah commented on runner.go:
```diff
@@ -107,6 +109,19 @@ func Run(options ...RunOption) {
+		var outputString string
+		for _, task := range tasks {
+			if len(task.Description) > 0 {
+				outputString = fmt.Sprintf("%s\n\t%s\t # %s", outputString, task.Name, task.Description)
```
Is the idea here that by adding the comment we are maintaining compatibility with fzf?

stansey replied:
I was hoping to maintain compatibility with fzf, but hashes quotes and semicolons were not enough to prevent taskrunner from trying to run every word in the comment as a task. I couldn't think of anything else that seemed promising. Would you like me to change the formatting?

berfarah commented on runner.go:
```diff
@@ -107,6 +109,19 @@ func Run(options ...RunOption) {
+	if describeTasks {
+		var outputString string
+		for _, task := range tasks {
+			if len(task.Description) > 0 {
```
I think we could clean up this conditional. Seems like we always know we want to display the task and _sometimes_ want to display the description. Could we compose this in such a way that we don't need an if/else?

stansey replied:
I wanted to conditionally include some punctuation and go doesn't have a ternary operator, unless I missed something. I removed the punctuation and conditional formatting.

stansey commented:
I added two commits with the fixes you requested. I removed the punctuation since it was more code and didn't buy more functionality other than visually separating the name and the description.  I also updated the pull request description. 
@berfarah Is there anything else you need to see before you can approve this change?

stansey commented:
Thanks for the tip about tabwriter. I'll try it out the next time I'm doing something like this.
